app/models/accion_en_snapshot.rb

```diff
@@ -1,13 +1,17 @@
 class AccionEnSnapshot < ActiveRecord::Base
   belongs_to :accion
   belongs_to :snapshot
-  
+
+  scope :con_nombre_accion, lambda {
+    select(:id, :accion_id, :nombre, :posicion, :visitas_recientes, :visitas_totales).joins(:accion)
+  }
+
   delegate :nombre, to: :accion
-  
+
   def self.crear_desde_dato(dato, i)
     create(accion: Accion.buscar_o_crear_desde_dato(dato),
-           posicion: i + 1,
+           posicion: dato['visitas'] || i + 1,
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.